src/test/java/org/jenkinsci/plugins/gitclient/GitClientCloneTest.java

```diff
@@ -328,11 +329,16 @@ public void test_clone_refspec() throws Exception {
                         }));
     }
 
+    private String randomSpace() {
+        return random.nextBoolean() ? " " : "";
+    }
+
     @Test
+    @Issue("JENKINS-70303") // spaces at start or end of refspec should be ignored
     public void test_clone_refspecs() throws Exception {
         List<RefSpec> refspecs = Arrays.asList(
-                new RefSpec("+refs/heads/master:refs/remotes/origin/master"),
-                new RefSpec("+refs/heads/1.4.x:refs/remotes/origin/1.4.x"));
+                new RefSpec(randomSpace() + "+refs/heads/master:refs/remotes/origin/master" + randomSpace()),
+                new RefSpec(randomSpace() + "+refs/heads/1.4.x:refs/remotes/origin/1.4.x" + randomSpace()));
```

Comment:
This may be exactly the wrong place for this type of test.  We generally prefer tests that are nearer to what a user would do.  This is changing an argument to the RefSpec constructor when the real end user provides a string into a data entry field on a Jenkins form or they provide a string to an option of a Pipeline `checkout` step.

Reply:
The test case did pass in my local.

Reply:
Which test case tests the Pipeline `checkout` step?

Reply:
I don't think that we'll be able to test Pipeline checkout from this plugin because it would require a dependency on the git plugin.  That would result in a circular dependency, since the git plugin depends on this plugin.  We'll need to test interactively.